packages/devtools-reps/src/object-inspector/utils.js

```diff
@@ -151,73 +233,112 @@ function sortProperties(properties) {
   });
 }
 
-function makeNumericalBuckets(props, bucketSize, parentPath, ownProperties) {
-  const numProperties = props.length;
+function makeNumericalBuckets(
+  propertiesNames: Array<string>,
+  bucketSize: number,
+  parent: ObjectInspectorItem,
+  ownProperties: Object
+) : Array<ObjectInspectorItem> {
+  const parentPath = parent.path;
+  const numProperties = propertiesNames.length;
   const numBuckets = Math.ceil(numProperties / bucketSize);
   let buckets = [];
   for (let i = 1; i <= numBuckets; i++) {
-    const bucketKey = `bucket${i}`;
+    const bucketKey = `${SAFE_PATH_PREFIX}bucket${i}`;
     const minKey = (i - 1) * bucketSize;
     const maxKey = Math.min(i * bucketSize - 1, numProperties);
     const bucketName = `[${minKey}..${maxKey}]`;
-    const bucketProperties = props.slice(minKey, maxKey);
+    const bucketProperties = propertiesNames.slice(minKey, maxKey);
 
     const bucketNodes = bucketProperties.map(name =>
       createNode(
+        parent,
         name,
         `${parentPath}/${bucketKey}/${name}`,
         ownProperties[name]
       )
     );
 
     buckets.push(
-      createNode(bucketName, `${parentPath}/${bucketKey}`, bucketNodes)
+      createNode(
+        parent,
+        bucketName,
+        `${parentPath}/${bucketKey}`,
+        bucketNodes,
+        NODE_TYPES.BUCKET
+      )
     );
   }
   return buckets;
 }
 
-function makeDefaultPropsBucket(props, parentPath, ownProperties) {
-  const userProps = props.filter(name => !isDefault({ name }));
-  const defaultProps = props.filter(name => isDefault({ name }));
+function makeDefaultPropsBucket(
+  propertiesNames: Array<string>,
+  parent: ObjectInspectorItem,
+  ownProperties: Object
```

Comment:
It's a dream of mine to fully type the grips... 
http://searchfox.org/mozilla-central/source/devtools/docs/backend/protocol.md

I've started doing this a couple times